### ISSUE TITLE
IProcessTransport.pid is incompletely documented

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3197]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3197|
|Type|defect|
|Created|2008-04-20 16:59:41Z|
|Branch|https://github.com/twisted/twisted/tree/pid-doc-3197-2|

After a process ends, its `pid` attribute is set to `None`.  The interface doesn't document this, or the type of `pid` while the process is running.

Attachments:

* [processtransport-pid.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c1b/c1b85a09d3b6f8251130e1865a37f7b8000b2bc7/f8363208ab5f60b71c166b8a85ae81e56383b59a.patch) (671 bytes) - added by exarkun on 2008-10-23 15:31:02Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3197 3197
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_pid_doc_3197_2 branches/pid-doc-3197-2
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1208710781000000 1208710781000000
changetime__1226811152000000 1226811152000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [24893]) Branching to 'pid-doc-3197'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Added the ```None``` docstring in r24894, but still need to know the 'type of pid while the process is running'.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

It's an int while it's not `None`.  It's worth mentioning that it is set to `None` _before_ `processEnded` is called, I think.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Updated the docstring in r25029. I manually merged forward, creating a new branch is too much for such a small change imo.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Small change, but it conflicts if I try to merge it with trunk.  I'm not sure what a "manual merge forward" is, but since it doesn't prevent conflicts I think this needs a new branch, or at least attach a patch with the resolved diff to the ticket.  Actually, I'll do that.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph set owner to @exarkun|
|-|-|

This is an improvement, it should be merged.  It's a bit awkwardly worded though - it implies that _after_ `processEnded` it called, `pid` is set to something different.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra commented|
|-|-|

(In [25316]) Branching to 'pid-doc-3197-2'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set owner to @exarkun<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [glyph](#note_9):
> This is an improvement, it should be merged.  It's a bit awkwardly worded though - it implies that _after_ `processEnded` it called, `pid` is set to something different.

That's what it implies but what should it say otherwise?

Created a new [branches/pid-doc-3197-2 branch](branches/pid-doc-3197-2 branch) and applied the patch. Assigning it to exarkun so he can possibly rephrase it and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [25324]) another approach

refs [#3197](https://github.com/twisted/twisted/issues/3197)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [25325]) s/C{/L{/

refs [#3197](https://github.com/twisted/twisted/issues/3197)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_16"></a>|@radix set owner to @exarkun|
|-|-|

I thought we don't use L{} for built-in types and None?

If we do, then merge. If we don't, change to C{} and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun commented|
|-|-|

Lately, I feel like `L{}` means fully-qualified Python name and `C{}` means evaluatable-ish expression.  So I guess I'll leave it as-is.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun set status to `closed`|
|-|-|

(In [25367]) Merge pid-doc-3197-2

Author: thijs, exarkun
Reviewer: radix
Fixes: [#3197](https://github.com/twisted/twisted/issues/3197)

Expand the interface documentation for `IProcessTransport.pid`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
